git/client.go

```diff
@@ -227,14 +247,13 @@ func (c *Client) UncommittedChangeCount(ctx context.Context) (int, error) {
 	return count, nil
 }
 
-func isGitSha(s string) bool {
-	shaRegex := regexp.MustCompile(`^[0-9a-fA-F]{7,40}$`)
-	ret := shaRegex.MatchString(s)
-	return ret
-}
-
 func (c *Client) Commits(ctx context.Context, baseRef, headRef string) ([]*Commit, error) {
-	args := []string{"-c", "log.ShowSignature=false", "log", "--pretty=format:%H,%s,%b", "--cherry", fmt.Sprintf("%s...%s", baseRef, headRef)}
+	// The formatting directive %x00 indicates that git should include the null byte as a separator.
+	// We use this because it is not a valid character to include in a commit message. Previously,
+	// commas were used here but when we Split on them, we would get incorrect results if commit titles
+	// happened to contain them.
+	// https://git-scm.com/docs/pretty-formats#Documentation/pretty-formats.txt-emx00em
+	args := []string{"-c", "log.ShowSignature=false", "log", "--pretty=format:%H%x00%s%x00%b%x00", "--cherry", fmt.Sprintf("%s...%s", baseRef, headRef)}
```

Comment:
**Why is the additional null byte separate after the body necessary?**

I only ask because we're not checking the size of the split line nor using that index in the code below.

Reply:
It might not be necessary anymore!

I added it because when the regex said `[\S\s]*` it would greedily consume the rest of the string including the newlines.

However, now that I've made it lazy with `?` I think it might stop at `$`

I'll check!

Reply:
Oh no, this doesn't work because for [commit bodies with newlines](https://github.com/cli/cli/pull/8768/files#diff-e0d3414f3f705f5bcd834cfa70ffcab2ac76b2a44b4f1a67331bacf5599722e9R579-R607), the regex without the final `%x00` will either lazily stop at `$` (i.e. the newline in the commit body) or greedily consume to the end of all the commit logs.